src/lib/fragments/AgGrid.react.js

```diff
@@ -949,19 +1073,63 @@ export default class DashAgGrid extends Component {
             this.rowTransaction(rowTransaction);
         }
 
+        let alignedGrids;
+        if (dashGridOptions) {
+            if ('alignedGrids' in dashGridOptions) {
+                alignedGrids = [];
+                const addGrid = (id) => {
+                    const strId = stringifyId(id);
+                    eventBus.on(this.props.id, strId, () => {
+                        this.setState({
+                            toggle: this.state.toggle
+                                ? !this.state.toggle
+                                : false,
+                        });
```

Comment:
Isn't this always going to be `false`? But anyway, if you DO want to modify state based on its previous data, you're supposed to use the functional form `prevState=>updates` since in theory React can batch these updates asynchronously.
```suggestion
                        this.setState((prevState) => {
                            toggle: !prevState.toggle
                        });
```
That said, if the purpose here is to trigger a rerender so the alignment will update, it might be better (a) name it to make that clear (`state.rerender`?), (b) make it an incrementing counter (`this.setState((prevState) => {rerender: prevState.rerender + 1})`) in case there are two back-to-back events that toggle on and back off and make it look like nothing happened.